### PR TITLE
Remove salted fish from pemmican recipe, add salted meat to a number of recipes

### PR DESCRIPTION
**Pull request by meelock**
#### Summary
Balance "Remove salted fish from pemmican recipe, add salted meat to a number of recipes and salted fish to a few recipes, remove duplicate salted fish entries from deluxe scrambled eggs."

#### Purpose of change
The recipe for pemmican allowed for salted fish, which is inadvisable according to [a source](https://grandpappy.org/rpemmica.htm). In addition I realized that many recipes that could have used salted meat, and there were some duplicate entries for salted fish in the recipes for deluxe scrambled eggs.

#### Describe the solution

This PR removes salted fish from the pemmican recipe and duplicate entries from deluxe scrambled eggs recipes. In addition this PR adds salted meat slices to the recipes for Canned Chowder, Deluxe Baked Beans, Deluxe Fried Rice, Deluxe Beans and Rice, and Meat Pie. Also added salted fish slices to Deluxe Baked Beans, Deluxe Fried Rice, Deluxe Beans and Rice, and Meat Pie.

#### Describe alternatives you've considered

Not adding salted meat slices to more foods, assuming that salting meat does something strange to foods. Or, adding salted meat slices to all foods that can use both dehydrated meat and fish as components. 

#### Testing

I quickly tested the addition, and saw no visible issues.

#### Additional context

I was originally planning to just add salted meat slices to the recipe for pemmican, but after seeing how few recipes salted meat slices can be used in, I decided to add them to more.

### CODE REVIEW

**Comment by actual-nh** [comment]
Sometimes you say "salted meat", and sometimes "vacuum-packed meat". 😕  Not all salted meat is vacuum-packed by any means, and I suspect the same is true the other way around.

**Comment by meelock** [comment]
Thank you, that has been fixed.

**Comment by souricelle** [comment]
Pemmican is not made from salted meat IRL, it is specifically only made with dehydrated meat.

**Comment by souricelle** [comment]
Salted beef is just cured, not dehydrated, right? I'm not sure it would work in pemmican. Pemmican relies on all moisture being removed from the meat so the tallow can make a seal on it. It basically has to be a crumbly dry powder before it will work.

**Comment by meelock** [comment]
Perhaps this could be solved with a recipe to dehydrate salted meat slices. I will remove salted meat slices from pemmican if others also think this doesn't make sense.

**Comment by souricelle** [comment]
I don't know if the meat would dehydrate the same way or whether it would be OK to make pemmican with that much salt in it. You only make pemmican the one way IRL, and the meat ain't salted.

**Comment by souricelle** [comment]
In a case like this I think the burden of proof is on you. If you can't find evidence for people making pemmican out of cured meat, you shouldn't try to add it.

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -2396,6 +2396,7 @@
         [ "meat_smoked", 2 ],
         [ "dry_fish", 2 ],
         [ "salted_fish", 2 ],
+        [ "meat_salted", 2 ],
```
Comment (souricelle) [inline data/json/recipes/recipe_food.json:2399]:
```suggestion

```

**Review by souricelle — CHANGES_REQUESTED** [review]

**Pull request merged**